packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -91,13 +91,32 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
                 values: quorumValues,
             },
         );
-        parsedUrl.pathname = parsedUrl.pathname.split("/").slice(0, -1).concat([documentId]).join("/");
+        /**
+         * Assume documentId is at end of url path.
+         * This is true for Routerlicious' and Tinylicious' documentUrl and deltaStorageUrl.
+         * Routerlicious and Tinylicious do not use documentId in storageUrl nor ordererUrl.
+         * TODO: Ideally we would be able to regenerate the resolvedUrl, rather than patching the current one.
```

Comment:
This should work for now. For long-term solution we should consider passing down the url resolver to regenerate the url instance with the new document id.